doc/ext-pake/proposed/spake2plus_proposal_draft.md

```diff
@@ -36,32 +37,28 @@ SPAKE2+ has an offline initialization step where both parties may share parties
     *   `psa_pake_set_user()` to input ProverID
     *   `psa_pake_set_peer()` to input VerifierID
     *   `psa_pake_set_role()` to set role as client/server
-    *   `psa_pake_input()` with *(new)* `PSA_PAKE_STEP_ADDITIONAL_DATA` step to input context (additional data).
-
-### Password, PBKDF Output and Verification value (W0, W1, L)
-
-SPAKE2+ requires PBKDF Output and Verification value (W0, W1, L) generated from password as below :
+*   Additional proposed API
+    *   `psa_pake_set_context()` to input context (additional data)
 
- `w0s || w1s = PBKDF(len(pw) || pw || len(A) || A || len(B) || B)`  
- `W0 = w0s mod p`  
- `W1 = w1s mod p`  
- `L = w1*P`
+### Registration
 
-Computing these values is considered Out of Scope for PAKE API as this a registration phase which could happen out of actual protocol flow.
+Propose a new API `psa_pake_registration()` which will take as input:
+* PAKE operation with user and peer IDs
+* PBKDF operation which is initialized with the key attributes, PBKDF parameters and password received OOB
+* key attributes for the output
 
-Since W0 & W1 and L serve as registartion records and verification value, these values can be treated as secrets.
+The function will compute `w0 || w1` if role is set as prover or `w0 || L` if role is set as verifier and store the result as a key.
 
-**Input methods** : Proposing two different approaches, protocol flow diagram has approach 2 mentioned below
-
-1.  Given secret nature of these values, we can use current `psa_pake_set_password_key()` to input these secrets.
-    1.  `W0||W1` as secret key on Prover side.
-    2.  `W0||L` as secret key on Verifier side.
-2.  Rename `psa_pake_set_password_key()` as `psa_pake_input_key()` to input mutiple secret keys and define steps for different keys
-    1.  For W0 and W1 on Prover side
-    2.  For W0 and L on Verifier side
+```
+psa_status_t psa_pake_registration(psa_pake_operation_t *pake, psa_key_derivation_operation_t *pbkdf, psa_key_attributes_t *attributes);
+```
 
 ### Key confirmation
 
 Key confirmation is part of the SPAKE2+ protocol. Current PSA Cryptography API 1.1 PAKE Extension only supports implicit key confirmation.
 
-*   New API `psa_pake_get_explicit_key()`  is required to provide keys with explicit key confirmation.
\ No newline at end of file
+*   New API `psa_pake_get_explicit_key()`  which will take as input the current PAKE operation and the attributes for the explicit key and return the key id of the explicit key.
```

Comment:
I wonder about the naming of this API, and by extension, the existing `psa_pake_get_implicit_key()`.

I think the API would be clearer if we called this `psa_pake_get_confirmed_key()` or `psa_pake_output_confirmed_key()`? (I like 'output' as this matches the key derivation operation, but I can see that we inject the password with a 'set' function.

As per #86, we should consider whether we want both a 'extract to key' API (which could extract only part of the output?) and a 'inject shared secret into a KDF' API (like the existing `psa_pake_get_implicit_key()` function).

This is both a big enough sub-topic, with implications for the existing beta API: I will raise a separate issue to have that discussion.